src/version/build_info.rs

```diff
@@ -0,0 +1,129 @@
+use std::str::FromStr;
+
+use chrono::{DateTime, NaiveDate, Utc};
+use conduit::{Request, Response};
+use semver;
+use serde_json;
+
+use app::RequestApp;
+use db::RequestTransaction;
+use models::{Rights, Version};
+use user::RequestUser;
+use util::{human, CargoResult, RequestUtils};
+use version::version_and_crate;
+use views::{EncodableMaxVersionBuildInfo, EncodableVersionBuildInfoUpload,
+            ParsedRustChannelVersion};
+
+use schema::*;
+
+#[derive(Clone, Identifiable, Associations, Debug, Queryable)]
+#[belongs_to(Version)]
+#[table_name = "build_info"]
+#[primary_key(version_id, rust_version, target)]
+/// Stores information about whether this version built on the specified Rust version and target.
+pub struct BuildInfo {
+    version_id: i32,
+    pub rust_version: String,
+    pub target: String,
+    pub passed: bool,
+}
+
+/// The columns to select from the `build_info` table. The table also stores `created_at` and
+/// `updated_at` metadata for each row, but we're not displaying those anywhere so we're not
+/// bothering to select them.
+pub const BUILD_INFO_FIELDS: (
```

Comment:
Are the `created_at` and `updated_at` fields ever used at all? Is it worth even having them in the first place?